### ISSUE TITLE
Import Error (ConfigParser) on Python 3.4.3 OS X 10.10.4

### ISSUE THREAD

**Issue by Balauue** [SAP/PyHDB#36]
I installed pyhdb via pip, got v.0.2.3 and included it as usual.

When including the library the following error comes up:

```
Python 3.4.3 (v3.4.3:9b73f1c3e601, Feb 23 2015, 02:52:03) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyhdb
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pyhdb/pyhdb/__init__.py", line 18, in <module>
    import ConfigParser
ImportError: No module named 'ConfigParser'
```

Do I have to install dependencies or such? Python finds a "configparser" module (lowercase), but not the ConfigParser. 

Thanks in advance!

**Comment by jaheba** [comment]
> **Note** The ConfigParser module has been renamed to configparser in Python 3. The 2to3 tool will automatically adapt imports when converting your sources to Python 3.

Ich vermute pyhdb importiert im Moment hart `ConfigParser`

**Comment by Balauue** [comment]
It is. For me it looks like the code was developed for 2.7 and not ported to 3, although python 3.4 compatibility is advertised.

In the meantime I pulled the current version 0.3.0 from git and ran setup.py, resulting in various syntax errors (see below):

```
mbp:pyhdb balauue$ python3 setup.py install
running install
running bdist_egg
running egg_info
creating pyhdb.egg-info
writing pyhdb.egg-info/PKG-INFO
writing top-level names to pyhdb.egg-info/top_level.txt
writing dependency_links to pyhdb.egg-info/dependency_links.txt
writing manifest file 'pyhdb.egg-info/SOURCES.txt'
reading manifest file 'pyhdb.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'pyhdb.egg-info/SOURCES.txt'
installing library code to build/bdist.macosx-10.6-intel/egg
running install_lib
running build_py
creating build
creating build/lib
creating build/lib/pyhdb
copying pyhdb/__init__.py -> build/lib/pyhdb
[...]
byte-compiling build/bdist.macosx-10.6-intel/egg/pyhdb/cursor.py to cursor.cpython-34.pyc
  File "build/bdist.macosx-10.6-intel/egg/pyhdb/cursor.py", line 37
    except TypeError, msg:
                    ^
SyntaxError: invalid syntax

[...]

byte-compiling build/bdist.macosx-10.6-intel/egg/pyhdb/protocol/parts.py to parts.cpython-34.pyc
  File "build/bdist.macosx-10.6-intel/egg/pyhdb/protocol/parts.py", line 555
    print 'writing lob buffers', row_lobs
                              ^
SyntaxError: Missing parentheses in call to 'print'

[...]

creating dist
creating 'dist/pyhdb-0.3.0-py3.4.egg' and adding 'build/bdist.macosx-10.6-intel/egg' to it
removing 'build/bdist.macosx-10.6-intel/egg' (and everything under it)
Processing pyhdb-0.3.0-py3.4.egg
creating /Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pyhdb-0.3.0-py3.4.egg
Extracting pyhdb-0.3.0-py3.4.egg to /Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pyhdb-0.3.0-py3.4.egg/pyhdb/cursor.py", line 37
    except TypeError, msg:
                    ^
SyntaxError: invalid syntax

  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pyhdb-0.3.0-py3.4.egg/pyhdb/protocol/parts.py", line 555
    print 'writing lob buffers', row_lobs
                              ^
SyntaxError: Missing parentheses in call to 'print'

Adding pyhdb 0.3.0 to easy-install.pth file

Installed /Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pyhdb-0.3.0-py3.4.egg
Processing dependencies for pyhdb==0.3.0
Finished processing dependencies for pyhdb==0.3.0
```

**Comment by ralhei** [comment]
Hi everyone,

it is true - although the code is announced as being 3.4 ready, the latest version(s) are actually not, simply because we had a lot of pressure to provide functionality for 2.7 without  being able to make it compliant for 3.4 (again). This is definitely an issue and should be addressed but it is not clear (to me) how quickly this can be achieved. 
@bsrdjan, @jarus : Should we remove the 3.4 compatibility flag for the moment until this issue is fixed?

**Comment by Balauue** [comment]
Hi,

thanks for clarification. I really hope for a quick 3.4 compliance. For you experts it is probably not even big effort as you know your code.

I tried using 2to3 tool and manually migrating obvious problems. Now I can include the library but a single SELECT query already yields exceptions:

```
  File "[...]/pyhdb/protocol/segments.py", line 150, in unpack_from
    raise error.parts[0].errors[0]
pyhdb.exceptions.DatabaseError: error while parsing protocol
```

**Comment by jarus** [comment]
Fixed by d2be2fb46cf85b64e9d670ea47c45baf38bc2a20.
